packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,413 @@ const double _kHandleSize = 22.0;
+        _maxWidth = renderBoxChild.size.width;
+      }
+    }
+
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    final List<Widget> items = <Widget>[
-      if (handleCut != null) FlatButton(child: Text(localizations.cutButtonLabel), onPressed: handleCut),
-      if (handleCopy != null) FlatButton(child: Text(localizations.copyButtonLabel), onPressed: handleCopy),
-      if (handlePaste != null) FlatButton(child: Text(localizations.pasteButtonLabel), onPressed: handlePaste),
-      if (handleSelectAll != null) FlatButton(child: Text(localizations.selectAllButtonLabel), onPressed: handleSelectAll),
-    ];
+    final List<Widget> items = <Widget>[];
+    if (widget.handleCut != null) {
+      items.add(FlatButton(
+        child: Text(localizations.cutButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handleCut();
+        },
+      ));
+    }
+    if (widget.handleCopy != null) {
+      items.add(FlatButton(
+        child: Text(localizations.copyButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handleCopy();
+        },
+      ));
+    }
+    if (widget.handlePaste != null) {
+      items.add(FlatButton(
+        child: Text(localizations.pasteButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handlePaste();
+        },
+      ));
+    }
+    if (widget.handleSelectAll != null) {
+      items.add(FlatButton(
+        child: Text(localizations.selectAllButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handleSelectAll();
+        },
+      ));
+    }
 
     // If there is no option available, build an empty widget.
     if (items.isEmpty) {
       return Container(width: 0.0, height: 0.0);
     }
 
-    return Material(
+    final Material child = Material(
+      key: _key,
       elevation: 1.0,
-      child: Container(
-        height: _kToolbarHeight,
-        child: Row(mainAxisSize: MainAxisSize.min, children: items),
+      child: AnimatedSize(
+        vsync: this,
+        // This duration was eyeballed on a Pixel 2 emulator running Android
+        // API 28.
+        duration: const Duration(milliseconds: 140),
+        child: _TextSelectionToolbarItems(
+          isAbove: widget.isAbove,
+          overflowOpen: _overflowOpen,
+          children: <Widget>[
+            // The navButton that shows and hides the overflow menu is the first
+            // child.
+            Material(
+              child: IconButton(
+                // TODO(justinmc): This should be an AnimatedIcon, but
+                // AnimatedIcons doesn't yet support arrow_back to more_vert.
+                // https://github.com/flutter/flutter/issues/51209
```

Comment:
In addition to this, the icon actually animates its position in the native menu.  How can I do that given that this button is positioned by a RenderBox?

Reply:
Actually, it seems my AnimatedSize doesn't work either now that I'm using a RenderBox.

Reply:
Do you have a gif for native animation? I am not sure what animation we should achieve here

Reply:
@justinmc

Reply:
Here it is on native Android:

![selection_menu_overflow_native_android_2](https://user-images.githubusercontent.com/389558/75579925-a9f14680-5a1b-11ea-8a1c-0085c1b23aca.gif)